# Asst.Prof.Syed Faisal Ali Programming Fundamentals-Fall 2019 Software Engineering Dated:29/11/2019


Question 1: Create a function to find the following: If the base of triangle is 3 cm long and its equilateral triangle and the radius of circle is 1.5 cm then find the area of triangle shaded.

In [3]:
def area(b,r):
    h=b*r/2
    return h
print(area(3,1.5))

2.25


Question 2: Create a function which can read a dictionary of your family members such as 5 members. 1 Abbu, 1 Ammi, 2 Brothers 1 Sister. Now feed this data in dictionary in terms of name and relations. The UDF will ask findrelation() in this you will enter Brother it will return the names of two brothers you have inserted. In case if the relation is not found it will return “Sorry the relation doesn’t exist in your family.”

In [5]:
def findrelation(d):
    dict = {'father':'wasim' , 'brother':'hamza, huzaifa' , 'sister':'khizra' , 'mother':'tanzila'}
    a = dict[d]
    return a
print(findrelation('brother'))

hamza, huzaifa


Question 3: Create a function to find the following: If the base of triangle is 5 cm long and its equilateral triangle and the radius of circle is 2.25 cm then find the area of triangle shaded.

In [8]:
import math
def function(b):
    h=b**2*((math.sqrt(3))/4)
    return(h)
print(function(5))

10.825317547305483


Question 4: Create a function that takes a list of random numbers from users and add only those which are even. If all the numbers are odd it will return sorry no even number found.

In [9]:
def ev(l):
    s=0
    for i in l:
        if i%2==0:
            print(i)
            s = s+i
            print("sum of even no:",s)
        else:
            print("No even found!")
print(ev([1,2,4,5,6,7,8,10]))

No even found!
2
sum of even no: 2
4
sum of even no: 6
No even found!
6
sum of even no: 12
No even found!
8
sum of even no: 20
10
sum of even no: 30
None


Question 5: Write a function which can take a list of numbers and it will return sorted list.

In [10]:
def function(l):
    return sorted(l)
print (function(['8','5','4','3']))

['3', '4', '5', '8']


Question 6: Write a function that will take the radius and return the perimeter and area of circle with 5% increment.

In [11]:
from math import pi
def area_circle(r):
    area=pi*(r**2)
    a = area+(area*0.05)
    print("Perimeter",2*pi*r)
    print("Area of Circle:",a)
    return ""
print(area_circle(6))

Perimeter 37.69911184307752
Area of Circle: 118.75220230569418



Question 7: Write a function that will take the strings as argument and return number of vowels and consonants.

In [14]:
#A

def function(ch):
    if(ch in 'A' or ch in 'a' or ch in 'E' or ch in 'e' or ch in 'I' or ch in 'i' or ch in 'O' or ch in 'o' or ch in 'U' or ch in 'u'):
        print(ch, "is a Vowel")
    else:
        print(ch, "is a Consonant")
        
print(function('Hamza'))    

Hamza is a Consonant
None


  Question 8: Write a function that will take length and breadth for a rectangle and return perimeter and area of rectangle with 8% increment.

In [24]:
from math import pi
def rec(l,b):
    area=l*b
    a=area+(area*0.08)
    print("area of rectangle:",a)
    print("perimeter:",2*(l+b))
rec(5,7)    

area of rectangle: 37.8
perimeter: 24


Question 9: Write a function that can take the numbers in strings. From string find which number is even and which one is odd. Save them in two different lists and generate the result.

In [26]:
a = "12345678"
b = 0
c = 0
ab = []
cd = []
for i in a:
    if int(i)%2==0:
        b += 1
        ab.append(i)
    else:
        c += 1
        cd.append(i)
print('Numbers of Even :',b)
print('Numbers of odd:',c)
print("List of even:",ab)
print("List of odd:",cd)

Numbers of Even : 4
Numbers of odd: 4
List of even: ['2', '4', '6', '8']
List of odd: ['1', '3', '5', '7']


Question 10: Write a function which will take the string from the user and return how many alphabets have been used in it and which alphabets are missing.

In [27]:
s = 'King'
n = 0
az = 'abcdefghijklmnopqrstuvwxyz'
for i in s:
    if i in az:
        az = az.replace(i,"")
        n += 1
print(n)
#
print(az)

3
abcdefhjklmopqrstuvwxyz


Question 11: Write a function that will take verbs in words and return a list of verbs with continuous tense by adding (ing) at the end of each verb.

In [29]:
def verb(s):
    if s[-1] == 'e':
        s = s.replace(s[-1],"")
        print("verb",s+"ing")
    else:
        print("verb",s+"ing")
verb('eat')

verb eating


Question 12: Make a function which can take two radius of circles and find the areas of it and subtract smaller one from larger one and tell the remaining area of circle.

In [33]:
from math import pi
def aq(r,rr):
    a = pi*(r**2)
    aa = pi*(rr**2)
    if a > aa:
        fa = (aa - a)
    else:
        fa = (a - aa)
    return abs(fa)
print(aq(10,20))

942.4777960769379


Question 13: Write a function that will take a string and calculate number of Upper case letters and lower case letters.

In [34]:
def op(s):
    a = 0
    b = 0
    u = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
    l = 'abcdefghijklmnopqrstuvwxyz'
    for i in s:
        if i in u:
            a = a+1
        if i in l:
            b = b+1
    print(b,a)
print(op("Hamza"))    

4 1
None


Question 14: Write a function which will take length and breadth of two rectangles. Subtract the smaller rectangle from the larger rectangle and return the area left behind.

In [36]:
def Zx(l,b,ll,bb):
    a = l*b
    aa = ll*bb
    if a > aa:
        fa = (aa - a)
    else:
        fa = (a - aa)
    return abs(fa)
print(Zx(3,2,8,10))

74


Question 15: Create a function that can add the fractions in series such as 1 to 8 = 1/8+1/7+1/6+1/5 …… ½ and return the result in fraction not in decimal.

In [37]:
from fractions import Fraction
i = 1 
f = 5
ff = 10
for ii in range(i,f):
    ff = ff + Fraction(1,ii)
print(ff)

145/12


Question 17: Write a function which will take a list of fruits names. The function will return how many alphabets are repetitive in the names of fruits and how many are unique letters.

In [40]:
s = 'Banana'
ss = ""
xx = 0
for i in s:
    n = s.count(i)
    if n > 1:
        xx = xx + 1
print('rep',xx)

rep 5


Question 18: Write a function that can take square length and radius of circle. Find the area of both and subtract the smallest shape from largest one and return the remaining shape area.

In [42]:
def hg(l,r):
    sa = l*l
    ca = 3.142*(r**2)
    print(sa)
    print(ca)
    if sa > ca:
        fa = (ca - sa)
    else:
        fa = (sa - ca)
    return abs(fa)  
print(hg(4,8))

16
201.088
185.088


Question 16: Write a function which will take height and base for a triangle and make it 5% deferment calculate area.Subtract both and find the remaining area.

In [1]:
def area(height,base,increment):
    a=(height*base)/2
    incr=5-100
    percent_Increase = ((incr/100)*100)
    print('Area of a rectangle is = ',a,percent_Increase)
area (5,8,5)
def area(height,base,increment):
    a=(height*base)/2
    incr=5-100
    percent_Increase = ((incr/100)*100)
    print('area of a rectangle is = ',a,percent_Increase)
area(18,56,5)
c=(20.0-504.0)
print('area left behind = ' ,c)

Area of a rectangle is =  20.0 -95.0
area of a rectangle is =  504.0 -95.0
area left behind =  -484.0
